torch/csrc/Storage.cpp

```diff
@@ -290,29 +290,50 @@ static void THPStorage_subclass_dealloc(PyObject* self) {
   Py_DECREF(type);
 }
 
-c10::intrusive_ptr<c10::StorageImpl> make_storage_impl(
-    c10::StorageImpl::use_byte_size_t use_byte_size,
-    c10::SymInt size_bytes,
-    c10::Allocator* allocator,
-    bool resizable,
-    c10::optional<int64_t> allocator_opt,
-    c10::optional<at::Device> device_opt) {
+static PyObject* THPStorage_pynew(
```

Comment:
Was moving this function really necessary? It makes it hard to read the diff

Reply:
> Was moving this function really necessary? It makes it hard to read the diff

I think it is necessary. These changes are mainly from reverting https://github.com/pytorch/pytorch/pull/100237. That pr extends storage creatation for custom storageImpl, but it doesn't consider other codes may need it like ``new_qtensor``. In my opinion, moving ``make_storage_impl`` to c10/core/StorageImpl.h is better for generality, @ezyang.

Reply:
I would agree with ezyang that the diff is pretty hard to read here.
You seems to be moving part of make_storage_impl into THPStorage_pynew below and the other half in c10?
It does change the order of quite a bit of the logic in here...